### PR TITLE
test: add benchmarking support for cdk

### PR DESCRIPTION
**Pull request by saikumarrs**
## Description of the change

Added a new benchmarking utility to compare CDK 2.0 and native implementations for transformations.
Also, fixed issues and refactored the logger module.

The output of the benchmark module:
- Operations
```
Destinations selected:  [ 'algolia', 'pinterest_tag' ]

==========================================
Destination - algolia - "0"
==========================================
Fastest:  "native"

"native" -  4.65 M ops/s ±636.5% (67328 runs sampled)

"CDK 2.0" -  1.45 k ops/s ±11.1% (1203 runs sampled)
-> "native" is faster by 3206.2 times to "CDK 2.0"

==========================================
Destination - algolia - "1"
==========================================
Fastest:  "native"

"native" -  4.44 M ops/s ±644.4% (65740 runs sampled)

"CDK 2.0" -  761.37 ops/s ±9.4% (873 runs sampled)
-> "native" is faster by 5827.8 times to "CDK 2.0"

==========================================
Destination - algolia - "2"
==========================================
Fastest:  "native"

"native" -  2.33 M ops/s ±472.0% (47867 runs sampled)

"CDK 2.0" -  2.27 k ops/s ±12.9% (1505 runs sampled)
-> "native" is faster by 1030.1 times to "CDK 2.0"

==========================================
Destination - algolia - "3"
==========================================
Fastest:  "native"

"native" -  2.31 M ops/s ±458.9% (47826 runs sampled)

"CDK 2.0" -  6.58 k ops/s ±29.4% (2563 runs sampled)
-> "native" is faster by 351.4 times to "CDK 2.0"

==========================================
Destination - algolia - "4"
==========================================
Fastest:  "native"

"native" -  1.96 M ops/s ±435.9% (44110 runs sampled)

"CDK 2.0" -  2.02 k ops/s ±16.9% (1421 runs sampled)
-> "native" is faster by 973.3 times to "CDK 2.0"

==========================================
Destination - algolia - "5"
==========================================
Fastest:  "native"

"native" -  4.62 M ops/s ±678.4% (67343 runs sampled)

"CDK 2.0" -  1.27 k ops/s ±18.5% (1128 runs sampled)
-> "native" is faster by 3633.2 times to "CDK 2.0"

==========================================
Destination - algolia - "6"
==========================================
Fastest:  "native"

"native" -  1.87 M ops/s ±426.1% (43099 runs sampled)

"CDK 2.0" -  2.43 k ops/s ±23.8% (1560 runs sampled)
-> "native" is faster by 767.9 times to "CDK 2.0"

==========================================
Destination - algolia - "7"
==========================================
Fastest:  "native"

"native" -  2.14 M ops/s ±416.5% (46012 runs sampled)

"CDK 2.0" -  8.79 k ops/s ±35.7% (2964 runs sampled)
-> "native" is faster by 243.2 times to "CDK 2.0"

==========================================
Destination - algolia - "8"
==========================================
Fastest:  "native"

"native" -  2.48 M ops/s ±484.7% (49540 runs sampled)

"CDK 2.0" -  7.54 k ops/s ±33.9% (2745 runs sampled)
-> "native" is faster by 329.7 times to "CDK 2.0"

==========================================
Destination - algolia - "9"
==========================================
Fastest:  "native"

"native" -  2.11 M ops/s ±444.1% (45335 runs sampled)

"CDK 2.0" -  12.49 k ops/s ±39.9% (3534 runs sampled)
-> "native" is faster by 168.8 times to "CDK 2.0"

==========================================
Destination - pinterest_tag - "0"
==========================================
Fastest:  "native"

"native" -  8.7 k ops/s ±26.0% (2949 runs sampled)

"CDK 2.0" -  212.31 ops/s ±2.6% (461 runs sampled)
-> "native" is faster by 41.0 times to "CDK 2.0"

==========================================
Destination - pinterest_tag - "1"
==========================================
Fastest:  "native"

"native" -  26.37 k ops/s ±41.3% (5127 runs sampled)

"CDK 2.0" -  399.42 ops/s ±10.5% (632 runs sampled)
-> "native" is faster by 66.0 times to "CDK 2.0"

==========================================
Destination - pinterest_tag - "2"
==========================================
Fastest:  "native"

"native" -  25.66 k ops/s ±34.0% (5055 runs sampled)

"CDK 2.0" -  420.36 ops/s ±9.1% (649 runs sampled)
-> "native" is faster by 61.0 times to "CDK 2.0"

==========================================
Destination - pinterest_tag - "3"
==========================================
Fastest:  "native"

"native" -  4.18 M ops/s ±612.6% (64197 runs sampled)

"CDK 2.0" -  33.01 k ops/s ±61.6% (5737 runs sampled)
-> "native" is faster by 126.6 times to "CDK 2.0"

==========================================
Destination - pinterest_tag - "4"
==========================================
Fastest:  "native"

"native" -  58.32 k ops/s ±80.8% (7633 runs sampled)

"CDK 2.0" -  1.12 k ops/s ±20.6% (1057 runs sampled)
-> "native" is faster by 52.3 times to "CDK 2.0"

==========================================
Destination - pinterest_tag - "5"
==========================================
Fastest:  "native"

"native" -  25.95 k ops/s ±39.0% (5069 runs sampled)

"CDK 2.0" -  443.11 ops/s ±12.4% (666 runs sampled)
-> "native" is faster by 58.6 times to "CDK 2.0"

==========================================
Destination - pinterest_tag - "6"
==========================================
Fastest:  "native"

"native" -  4.17 M ops/s ±610.7% (64117 runs sampled)

"CDK 2.0" -  33.88 k ops/s ±62.6% (5818 runs sampled)
-> "native" is faster by 123.1 times to "CDK 2.0"

==========================================
Destination - pinterest_tag - "7"
==========================================
Fastest:  "native"

"native" -  4.1 M ops/s ±595.9% (62929 runs sampled)

"CDK 2.0" -  21.18 k ops/s ±50.3% (4600 runs sampled)
-> "native" is faster by 193.5 times to "CDK 2.0"

==========================================
Destination - pinterest_tag - "8"
==========================================
Fastest:  "native"

"native" -  561.86 k ops/s ±229.8% (23561 runs sampled)

"CDK 2.0" -  3.31 k ops/s ±24.3% (1818 runs sampled)
-> "native" is faster by 169.8 times to "CDK 2.0"

==========================================
Destination - pinterest_tag - "9"
==========================================
Fastest:  "native"

"native" -  25.67 k ops/s ±37.4% (5067 runs sampled)

"CDK 2.0" -  418.61 ops/s ±10.3% (647 runs sampled)
-> "native" is faster by 61.3 times to "CDK 2.0"

==========================================
Destination - pinterest_tag - "10"
==========================================
Fastest:  "native"

"native" -  28.63 k ops/s ±36.5% (5356 runs sampled)

"CDK 2.0" -  837.22 ops/s ±17.3% (915 runs sampled)
-> "native" is faster by 34.2 times to "CDK 2.0"

==========================================
Destination - pinterest_tag - "11"
==========================================
Fastest:  "native"

"native" -  30.52 k ops/s ±39.7% (5527 runs sampled)

"CDK 2.0" -  448.89 ops/s ±6.5% (671 runs sampled)
-> "native" is faster by 68.0 times to "CDK 2.0"

==========================================
Destination - pinterest_tag - "12"
==========================================
Fastest:  "native"

"native" -  23.67 k ops/s ±25.4% (4865 runs sampled)

"CDK 2.0" -  861.18 ops/s ±20.8% (928 runs sampled)
-> "native" is faster by 27.5 times to "CDK 2.0"

==========================================
Destination - pinterest_tag - "13"
==========================================
Fastest:  "native"

"native" -  29.36 k ops/s ±34.2% (5402 runs sampled)

"CDK 2.0" -  761.37 ops/s ±17.9% (873 runs sampled)
-> "native" is faster by 38.6 times to "CDK 2.0"

==========================================
Destination - pinterest_tag - "14"
==========================================
Fastest:  "native"

"native" -  49.31 k ops/s ±73.3% (7022 runs sampled)

"CDK 2.0" -  926.44 ops/s ±24.7% (963 runs sampled)
-> "native" is faster by 53.2 times to "CDK 2.0"
```

- Memory
```
Destinations selected:  [ 'algolia', 'pinterest_tag' ]

==========================================
Destination - algolia - "0"
==========================================
Memory intensive:  "CDK 2.0"

"native" -  3.05 kB ±0.1% (32 runs sampled)
-> "CDK 2.0" consumed 1.1 times memory compared to "native"

"CDK 2.0" -  3.28 kB ±0.9% (32 runs sampled)

==========================================
Destination - algolia - "1"
==========================================
Memory intensive:  "native"

"native" -  461 B ±0.4% (33 runs sampled)

"CDK 2.0" -  458 B ±0.5% (32 runs sampled)
-> "native" consumed 1.0 times memory compared to "CDK 2.0"

==========================================
Destination - algolia - "2"
==========================================
Memory intensive:  "CDK 2.0"

"native" -  2.72 kB ±0.4% (34 runs sampled)
-> "CDK 2.0" consumed 1.0 times memory compared to "native"

"CDK 2.0" -  2.73 kB ±1.0% (32 runs sampled)

==========================================
Destination - algolia - "3"
==========================================
Memory intensive:  "native"

"native" -  2.57 kB ±0.2% (34 runs sampled)

"CDK 2.0" -  719 B ±1.0% (31 runs sampled)
-> "native" consumed 3.6 times memory compared to "CDK 2.0"

==========================================
Destination - algolia - "4"
==========================================
Memory intensive:  "native"

"native" -  1.34 kB ±0.0% (35 runs sampled)

"CDK 2.0" -  291 B ±0.2% (31 runs sampled)
-> "native" consumed 4.6 times memory compared to "CDK 2.0"

==========================================
Destination - algolia - "5"
==========================================
Memory intensive:  "CDK 2.0"

"native" -  205 B ±0.0% (33 runs sampled)
-> "CDK 2.0" consumed 13.6 times memory compared to "native"

"CDK 2.0" -  2.79 kB ±2.1% (30 runs sampled)

==========================================
Destination - algolia - "6"
==========================================
Memory intensive:  "CDK 2.0"

"native" -  2.13 kB ±0.2% (33 runs sampled)
-> "CDK 2.0" consumed 1.5 times memory compared to "native"

"CDK 2.0" -  3.16 kB ±0.6% (31 runs sampled)

==========================================
Destination - algolia - "7"
==========================================
Memory intensive:  "native"

"native" -  3.43 kB ±0.0% (32 runs sampled)

"CDK 2.0" -  2.65 kB ±0.0% (28 runs sampled)
-> "native" consumed 1.3 times memory compared to "CDK 2.0"

==========================================
Destination - algolia - "8"
==========================================
Memory intensive:  "native"

"native" -  3.4 kB ±0.0% (33 runs sampled)

"CDK 2.0" -  1.21 kB ±1.1% (28 runs sampled)
-> "native" consumed 2.8 times memory compared to "CDK 2.0"

==========================================
Destination - algolia - "9"
==========================================
Memory intensive:  "native"

"native" -  1.69 kB ±0.1% (28 runs sampled)

"CDK 2.0" -  204 B ±0.0% (29 runs sampled)
-> "native" consumed 8.3 times memory compared to "CDK 2.0"

==========================================
Destination - pinterest_tag - "0"
==========================================
Memory intensive:  "CDK 2.0"

"native" -  945 B ±0.6% (33 runs sampled)
-> "CDK 2.0" consumed 3.1 times memory compared to "native"

"CDK 2.0" -  2.92 kB ±1.0% (24 runs sampled)

==========================================
Destination - pinterest_tag - "1"
==========================================
Memory intensive:  "native"

"native" -  1.42 kB ±1.0% (30 runs sampled)

"CDK 2.0" -  928 B ±1.2% (23 runs sampled)
-> "native" consumed 1.5 times memory compared to "CDK 2.0"

==========================================
Destination - pinterest_tag - "2"
==========================================
Memory intensive:  "native"

"native" -  2.9 kB ±1.2% (31 runs sampled)

"CDK 2.0" -  1.87 kB ±1.2% (26 runs sampled)
-> "native" consumed 1.5 times memory compared to "CDK 2.0"

==========================================
Destination - pinterest_tag - "3"
==========================================
Memory intensive:  "CDK 2.0"

"native" -  1.23 kB ±0.1% (34 runs sampled)
-> "CDK 2.0" consumed 2.4 times memory compared to "native"

"CDK 2.0" -  2.97 kB ±0.4% (32 runs sampled)

==========================================
Destination - pinterest_tag - "4"
==========================================
Memory intensive:  "native"

"native" -  3.26 kB ±0.2% (33 runs sampled)

"CDK 2.0" -  201 B ±0.0% (29 runs sampled)
-> "native" consumed 16.3 times memory compared to "CDK 2.0"

==========================================
Destination - pinterest_tag - "5"
==========================================
Memory intensive:  "CDK 2.0"

"native" -  644 B ±1.0% (33 runs sampled)
-> "CDK 2.0" consumed 3.9 times memory compared to "native"

"CDK 2.0" -  2.51 kB ±1.2% (26 runs sampled)

==========================================
Destination - pinterest_tag - "6"
==========================================
Memory intensive:  "CDK 2.0"

"native" -  1.27 kB ±0.1% (34 runs sampled)
-> "CDK 2.0" consumed 2.1 times memory compared to "native"

"CDK 2.0" -  2.65 kB ±0.0% (33 runs sampled)

==========================================
Destination - pinterest_tag - "7"
==========================================
Memory intensive:  "CDK 2.0"

"native" -  1.28 kB ±0.0% (34 runs sampled)
-> "CDK 2.0" consumed 2.2 times memory compared to "native"

"CDK 2.0" -  2.8 kB ±0.3% (32 runs sampled)

==========================================
Destination - pinterest_tag - "8"
==========================================
Memory intensive:  "native"

"native" -  4.54 kB ±0.5% (33 runs sampled)

"CDK 2.0" -  2.73 kB ±0.9% (30 runs sampled)
-> "native" consumed 1.7 times memory compared to "CDK 2.0"

==========================================
Destination - pinterest_tag - "9"
==========================================
Memory intensive:  "CDK 2.0"

"native" -  323 B ±0.4% (32 runs sampled)
-> "CDK 2.0" consumed 2.9 times memory compared to "native"

"CDK 2.0" -  935 B ±1.1% (25 runs sampled)

==========================================
Destination - pinterest_tag - "10"
==========================================
Memory intensive:  "CDK 2.0"

"native" -  357 B ±0.5% (31 runs sampled)
-> "CDK 2.0" consumed 3.5 times memory compared to "native"

"CDK 2.0" -  1.26 kB ±1.2% (26 runs sampled)

==========================================
Destination - pinterest_tag - "11"
==========================================
Memory intensive:  "CDK 2.0"

"native" -  201 B ±0.0% (31 runs sampled)
-> "CDK 2.0" consumed 10.7 times memory compared to "native"

"CDK 2.0" -  2.14 kB ±0.7% (26 runs sampled)

==========================================
Destination - pinterest_tag - "12"
==========================================
Memory intensive:  "CDK 2.0"

"native" -  1.2 kB ±1.1% (32 runs sampled)
-> "CDK 2.0" consumed 1.9 times memory compared to "native"

"CDK 2.0" -  2.31 kB ±0.7% (26 runs sampled)

==========================================
Destination - pinterest_tag - "13"
==========================================
Memory intensive:  "CDK 2.0"

"native" -  652 B ±0.3% (33 runs sampled)
-> "CDK 2.0" consumed 2.1 times memory compared to "native"

"CDK 2.0" -  1.38 kB ±0.7% (26 runs sampled)

==========================================
Destination - pinterest_tag - "14"
==========================================
Memory intensive:  "native"

"native" -  208 B ±0.0% (33 runs sampled)

"CDK 2.0" -  157 B ±0.1% (25 runs sampled)
-> "native" consumed 1.3 times memory compared to "CDK 2.0"
```

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

N/A

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: benchmark/index.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+  const results = {};
+  const suite = new Benchmark(suiteName, "Operations");
+
+  suite.add("native", async function() {
+    try {
+      await versionedRouter.handleV0Destination(
+        nativeDestHandlers[intg].process,
+        intg,
+        input,
+        "proc"
```
Comment (koladilip) [inline benchmark/index.js:127]: There are already constants for this right?

File: benchmark/index.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+      );
+    } catch (err) {
+      // logger.info(err);
+      // Do nothing
+    }
+  });
+
+  suite.add("CDK 2.0", async function() {
+    try {
+      await cdkV2Handler.processCdkV2Workflow(intg, input, "proc");
```
Comment (koladilip) [inline benchmark/index.js:137]: There are already constants for 'proc' right?

**Review by koladilip — COMMENTED** [review]
LGTM

File: benchmark/index.js (hunk 1 above)
Comment (saikumarrs) [reply benchmark/index.js]: Yeah, I was lazy ;) . Fixed it.

File: benchmark/index.js (hunk 2 above)
Comment (saikumarrs) [reply benchmark/index.js]: Fixed it.

**Review by koladilip — APPROVED** [review]

File: benchmark/index.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+      );
+    } catch (err) {
+      // logger.info(err);
+      // Do nothing
+    }
+  });
+
+  suite.add("CDK 2.0", async function() {
+    try {
+      await cdkV2Handler.processCdkV2Workflow(
```
Comment (koladilip) [inline benchmark/index.js:138]: As we discussed please create workflow engine instance before executing the benchmarking

Comment (saikumarrs) [reply benchmark/index.js]: Addressed.

**Review by koladilip — APPROVED** [review]

File: v0/destinations/ga4/transform.js [hunk 4]
```diff
@@ -324,6 +325,11 @@ const responseBuilder = (message, { Config }) => {
     };
   }
 
+  const integrationsObj = getIntegrationsObj(message, "ga4");
```
Comment (sanpj2292) [inline v0/destinations/ga4/transform.js:328]: [Small check]: Should this change go as part of this PR ?

**Review by sanpj2292 — DISMISSED** [review]

File: benchmark/index.js [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+        logger.info(
+          impl,
+          results[impl].stats.n,
+          results[impl].stats.mean.toFixed(4)
+        );
+        logger.info(
+          `"${result.end.name}" is faster by ${(
+            result.end.stats.n / results[impl].stats.n
+          ).toFixed(1)} times to "${impl}"`
+        );
```
Comment (koladilip) [inline benchmark/index.js:180]:
```suggestion
       if(result.end.name !==  results[impl].name){
          logger.info(
            `"${result.end.name}" is faster by ${(
              result.end.stats.n / results[impl].stats.n
            ).toFixed(1)} times to "${impl}"`
          );
        }
```

File: benchmark/index.js [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+      let tcDesc;
+      if (
+        "description" in curTcData &&
+        "input" in curTcData &&
+        "output" in curTcData
+      ) {
+        tcInput = curTcData.input;
+        tcDesc = `Source - ${src} - "${curTcData.description}"`;
+      } else {
+        tcInput = curTcData;
```
Comment (krishna2020) [inline benchmark/index.js:169]: For the test cases where we don't have new template of input,output in same file, shouldn't we take `curTcData .message` ?

File: benchmark/index.js [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+
+async function initializeHandlers() {
+  for (const idx in destinationsList) {
+    const dest = destinationsList[idx];
+
+    // Native destination handler
+    nativeDestHandlers[dest] = versionedRouter.getDestHandler(
+      "v0",
+      dest
+    ).process;
```
Comment (krishna2020) [inline benchmark/index.js:87]: what about router transformation flow benchmarking ? `processRouterDest`

File: v0/destinations/ga4/transform.js (hunk 4 above)
Comment (saikumarrs) [reply v0/destinations/ga4/transform.js]: No. Thanks for pointing that out. Reverted it.

File: benchmark/index.js (hunk 5 above)
Comment (saikumarrs) [reply benchmark/index.js]: Fixed this as part of the latest updates.

File: benchmark/index.js (hunk 7 above)
Comment (saikumarrs) [reply benchmark/index.js:87]: At this point, we are only supporting processor transformation. The router support will be added as a next step.

**Review by sanpj2292 — APPROVED** [review]

**Review by koladilip — APPROVED** [review]

**Review by krishna2020 — APPROVED** [review]

**Pull request merged**